### PR TITLE
Allow single account command in transaction inputs

### PR DESCRIPTION
**Pull request by azarovh**
- Main motivation for this PR is to allow only one authorizable account command per tx inputs. To easily control that I isolated all commands into a separate type so that they can be easily checked for uniqueness.
- Also this PR allows empty outputs in transaction that has input commands because some commands do not produce output (lock supply, freeze, change authority) thus it won't require user to use fake outputs

### CODE REVIEW

File: chainstate/tx-verifier/src/transaction_verifier/input_output_policy/purposes_check.rs [hunk 1]
```diff
@@ -159,66 +158,19 @@ pub fn check_tx_inputs_outputs_purposes(
...
-    );
+    // if provided account command must be unique among other inputs
+    let account_commands_count = tx
+        .inputs()
+        .iter()
+        .filter(|input| match input {
+            TxInput::Utxo(_) | TxInput::Account(..) => false,
+            TxInput::AccountCommand(..) => true,
+        })
+        .count();
```
Comment (TheQuantumPhysicist) [inline chainstate/tx-verifier/src/transaction_verifier/input_output_policy/purposes_check.rs:169]: Worth it... geez it was a nightmare of complexity.

**Review by TheQuantumPhysicist — APPROVED** [review]

File: chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs [hunk 2]
```diff
@@ -480,15 +482,14 @@ fn calculate_required_fee_burn(
...
-                | AccountOp::UnfreezeToken(_) => false,
-                AccountOp::ChangeTokenAuthority(_, _) => true,
+            TxInput::Utxo(_) | TxInput::Account(_) => false,
+            TxInput::AccountCommand(_, account_op) => match account_op {
+                AccountCommand::MintTokens(_, _)
+                | AccountCommand::UnmintTokens(_)
+                | AccountCommand::LockTokenSupply(_)
+                | AccountCommand::FreezeToken(_, _)
+                | AccountCommand::UnfreezeToken(_) => false,
+                AccountCommand::ChangeTokenAuthority(_, _) => true,
```
[the remark this answers is not in the document]
Comment (TheQuantumPhysicist) [reply chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs:492]: Evan though I merged, @azarovh , don't make this stop you from considering this comment. I merged for (obvious) logistic reasons.

[the remark this answers is not in the document]
Comment (azarovh) [reply chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs:492]: Though in general I agree, I'd not spend time on this. I'm planning to remove the whole module in the nearest future by moving to accumulator. And I don't see `get_account_command` useful there

[the remark this answers is not in the document]
Comment (iljakuklic) [reply chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs:492]: > Though in general I agree, I'd not spend time on this. I'm planning to remove the whole module in the nearest future by moving to accumulator. And I don't see `get_account_command` useful there

Sounds good

**Pull request merged**